benchmark/clBench/clblast/blas2/spr2.cpp

```diff
@@ -0,0 +1,172 @@
+      (mem_readWriteA + mem_readXreadY) * sizeof(scalar_t);
+  state.counters["bytes_processed"] = tot_mem_processed;
+
+  ExecutorType& ex = *executorPtr;
+
+  const clblast::Triangle triangle =
+      uplo == 'u' ? clblast::Triangle::kUpper : clblast::Triangle::kLower;
+
+  const int m_size = size * size;
+  const int v_x_size = 1 + (size - 1) * std::abs(incX);
+  const int v_y_size = 1 + (size - 1) * std::abs(incY);
+
+  // Input matrix/vector, output vector.
+  std::vector<scalar_t> m_a =
+      blas_benchmark::utils::random_data<scalar_t>(m_size);
+  std::vector<scalar_t> v_x =
+      blas_benchmark::utils::random_data<scalar_t>(v_x_size);
+  std::vector<scalar_t> v_y =
+      blas_benchmark::utils::random_data<scalar_t>(v_y_size);
+
+  MemBuffer<scalar_t> m_a_gpu(executorPtr, m_a.data(),
+                              static_cast<size_t>(m_size));
+  MemBuffer<scalar_t> v_x_gpu(executorPtr, v_x.data(),
+                              static_cast<size_t>(v_x_size));
+  MemBuffer<scalar_t> v_y_gpu(executorPtr, v_y.data(),
+                              static_cast<size_t>(v_y_size));
+
+  clblast::Layout layout = clblast::Layout::kColMajor;
+#ifdef BLAS_VERIFY_BENCHMARK
+  // Run a first time with a verification of the results
+  std::vector<scalar_t> x_ref = v_x;
+  std::vector<scalar_t> y_ref = v_y;
+  std::vector<scalar_t> m_a_ref = m_a;
+  reference_blas::spr2<scalar_t>(&uplo, size, alpha, x_ref.data(), incX,
+                                 y_ref.data(), incY, m_a_ref.data());
+  std::vector<scalar_t> m_a_temp = m_a;
+  {
+    MemBuffer<scalar_t> m_a_temp_gpu(executorPtr, m_a_temp.data(),
+                                     static_cast<size_t>(m_size));
+
+    cl_event event;
+    clblast::Spr2<scalar_t>(layout, triangle, size, alpha, v_x_gpu.dev(), 0,
+                            incX, v_y_gpu.dev(), 0, incY, m_a_temp_gpu.dev(), 0,
+                            executorPtr->_queue(), &event);
+    CLEventHandler::wait(event);
+  }
+
+  std::ostringstream err_stream;
+  if (!utils::compare_vectors(m_a_temp, m_a_ref, err_stream, "")) {
+    const std::string& err_str = err_stream.str();
+    state.SkipWithError(err_str.c_str());
+    *success = false;
+  };
+#endif
+
+  auto blas_method_def = [&]() -> std::vector<cl_event> {
+    cl_event event;
+    clblast::Spr2<scalar_t>(layout, triangle, size, alpha, v_x_gpu.dev(), 0,
+                           incX, v_y_gpu.dev(), 0, incY, m_a_gpu.dev(), 0,
+                           executorPtr->_queue(), &event);
+    CLEventHandler::wait(event);
+    return {event};
+  };
+
+  // Warmup
+  blas_benchmark::utils::warmup(blas_method_def);
+
+  blas_benchmark::utils::init_counters(state);
+
+  // Measure
+  for (auto _ : state) {
+    // Run
+    std::tuple<double, double> times =
+        blas_benchmark::utils::timef(blas_method_def);
+
+    // Report
+    blas_benchmark::utils::update_counters(state, times);
+  }
+
+  state.SetBytesProcessed(state.iterations() * tot_mem_processed);
```

Comment:
These two lines would also need to be changed to `state.SetBytesProcessed(state.iterations() * state.counters["bytes_processed"]);` and `state.SetItemsProcessed(state.iterations() * state.counters["n_fl_ops"]`.